src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,42 @@
+from typing import Mapping
+
+from sentry.sentry_metrics.metrics_interface import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+
+
+class KafkaGenericMetricsBackend(GenericMetricsBackend):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def counter(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
```

Comment:
This is quite an opinionated interface. Why is org and project ID needed at all? Seems like it can't be used for any use cases that aren't within the context of a single org and project then?

Reply:
This is part of the back-end schema so I don't think we should drop these.

Reply:
That's because the backend schema is a multi-tenant system and we need to keep customer data separate from other customers. For Sentry, this seems like it should always be org_id = 1 and project_id = 1. Why asking the user to pass it every time here?

Reply:
This wouldn't be the Sentry org_id and project_id since this would be customer data. @onewland am I missing anything?

Reply:
So just to clarify, the metrics platform is for aggregating customer data received during ingest only? I think this restriction wasn't super clear to me before, and I assumed it was valid to use metrics for anything at all.

Reply:
I think from both a query efficiency (the back-end tables are ordered first by org_id and project_id) and cost accounting perspective we should aim to include customer project IDs and org IDs where they make sense, and we can pass Sentry org IDs and project IDs where they make sense.

For example, on the escalating issues project, we're trying to keep a count of error by group ID on ingest. This has an obvious correspondence to a customer's project_id and org_id, and therefore that data should propagate through. 

If we wanted to track something like the number of times a celery task is executed, and this is independent from customer data/behavior, then I think it would make more sense to do org_id = 1 and project_id = 1. 

Lyn, how would you feel if these parameters were optional and the backends did some resolution if org_id/project_id aren't passed?

Reply:
We can't just hardcode org_id 1 and and project_id 1 though. Is there a plan for how these values will get resolved?

Does this interface make using Relay a non-starter? Or does there need to be other changes there to make this happen? The implication of this interface seems to be that a project can send data onto some other org/project not just itself. I am guessing this is not something generally allowed in Relay as it might have all kinds of implications in terms of security, quotas, etc.

Reply:
What changes are planned on Relay-side? Without any, you won't be able to implement this interface.

Reply:
right, so just to clarify - the internal Relay endpoint mentioned above would allow us to skip authenticating as Sentry, and would allow us to send customer org_id and project_id to Relay via this interface